File: Level1/Kth Number.cpp
/*
	https://programmers.co.kr/learn/courses/30/lessons/42748
*/
#include <vector>
#include <algorithm>

using namespace std;

enum { START_INDEX, END_INDEX, K_INDEX };

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
	vector<int> answer;

	for (const vector<int>& command : commands)
	{
		const vector<int> COMMAND{ command[START_INDEX] - 1, command[END_INDEX], command[K_INDEX] - 1 };
		vector<int> cuttedArray(array.begin() + COMMAND[START_INDEX], array.begin() + COMMAND[END_INDEX]);

		sort(cuttedArray.begin(), cuttedArray.end());

		answer.emplace_back(cuttedArray[COMMAND[K_INDEX]]);
	}

	return answer;
}

File: Java/Level1/Kth Number.java
/*
 * https://school.programmers.co.kr/learn/courses/30/lessons/42748?language=java
 */

import java.util.Arrays;

class Solution {

    public int[] cutArray(int[] array, int begin, int end)
    {
        final int size = end - begin + 1;
        int[] result = new int[size];

        for(int i = 0; i < size; ++i)
        {
            result[i] = array[begin + i];
        }

        return result;
    }

    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];

        for(int i = 0; i < commands.length; ++i)
        {
            int[] cuttedArray = cutArray(array, commands[i][0] - 1, commands[i][1] - 1);

            Arrays.sort(cuttedArray);
            answer[i] = cuttedArray[commands[i][2] - 1];
        }

        return answer;
    }
}